File: LeetCode/cpp/26_Remove_Duplicates_from_Sorted_Array.cpp
class Solution {
public:
	int removeDuplicates(vector<int>& nums) {
		unordered_map<int, int> um;
		int answer = 0;

		for (int i = 0; i < nums.size(); i++) {
			if (um[nums[i]] == 0) {
				um[nums[i]]++;
				continue;
			}

			nums[i] = 101;
		}

		sort(nums.begin(), nums.end());

		answer = um.size();

		return answer;
	}
};

File: LeetCode/java/26_Remove_Duplicates_from_Sorted_Array.java
import java.util.Arrays;
import java.util.HashMap;

class Solution {
    public int removeDuplicates(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int answer = 0;

        for (int i = 0; i < nums.length; i++) {
            if (!map.containsKey(nums[i])) {
                map.put(nums[i], 1);
                continue;
            }

            nums[i] = 101;
        }

        Arrays.sort(nums);

        answer = map.size();

        return answer;
    }
}